### ISSUE TITLE
createLocalTracks and mergeDefaultOptions doesn't clone deeply

### ISSUE THREAD

**Issue by vas11yev1work** [livekit/client-sdk-js#897]
### Describe the bug

`mergeDefaultOptions` inside `createLocalTracks` gets `options` as a parameter and change it. It happens because `options` can have nested objects. But in `mergeDefaultOptions` you do the following:

```typescript
const opts: CreateLocalTracksOptions = {
  ...options,
};
``` 

Is it intended or does it need to be fixed?

### Reproduction

You can see it here https://codesandbox.io/s/cocky-ritchie-ry68d9

In App.vue file

I created a `previewOptions` computed and pass it in `createLocalTracks`

### System Info

```shell
Windows 11 22621
x64 12th Gen Intel(R) Core(TM) i5-12500H
16 GB RAM
Node: 18.18.2
Yarn: 1.22.19
npm: 9.8.1
Chrome: 118.0.5993.89
"livekit-client": "^1.14.1"
```

### Severity

serious, but I can work around it

**Comment by lukasIO** [comment]
Thanks for the report! I think we should fix this. 
Do you want to open a PR that serializes and de-serializes the `CreateLocalTrackOptions`?

**Comment by vas11yev1work** [comment]
@lukasIO yes, I do
But what do you think is the best way to do it? I think to create a deepClone function (or this function has already been created). Or just use JSON namespace. But I don't sure about correctness and performance of this solution. Maybe use lodash or something similar?
I haven't delved into the specifics of the project, so I don't know how to do it better

**Comment by lukasIO** [comment]
I think for this case, the best way would be to use `structuredClone` if it's available in the browser and fallback to `JSON.parse(JSON.stringify())` if it's not available. 
If you do this, it would be great if you could add a unit test for it as well! 
https://developer.mozilla.org/en-US/docs/Web/API/structuredClone

**Comment by lukasIO** [comment]
closed via #906, thanks @vas11yev1work !
